### PR TITLE
apiserver: return BadRequest 400 for invalid query params

### PR DESCRIPTION
**Pull request by sttts**
Fixes https://github.com/kubernetes/kubernetes/issues/39730
Replaces #40773.

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/endpoints/handlers/rest.go [hunk 1]
```diff
@@ -903,6 +908,7 @@ func DeleteResource(r rest.GracefulDeleter, allowsOptions bool, scope RequestSco
 		if allowsOptions {
 			body, err := readBody(req)
 			if err != nil {
+				err = errors.NewBadRequest(err.Error())
```
Comment (cheftako) [inline staging/src/k8s.io/apiserver/pkg/endpoints/handlers/rest.go]: I'm not convinced we can claim any failure to read the body is a result of a malformed request. If we cannot then we shouldn't be blanket returning a 400 here.

**Comment by smarterclayton** [comment]
500 is more appropriate for reading the input body, yes.

File: staging/src/k8s.io/apiserver/pkg/endpoints/handlers/rest.go (hunk 1 above)
Comment (sttts) [inline staging/src/k8s.io/apiserver/pkg/endpoints/handlers/rest.go]: fixed

**Pull request merged**